### PR TITLE
Implement WebSockets for Profile download

### PR DESCRIPTION
**Pull request by kaankabalak**
Fixes miniohq/engineering#493
Fixes miniohq/engineering#494

### CODE REVIEW

**Comment by kaankabalak** [comment]
Seems like the integration tests are failing, will take a look at it. Marking the PR as WIP for now.

```
=== CONT  TestStartProfiling
    profiling_test.go:79: 
        	Error Trace:	profiling_test.go:79
        	Error:      	Not equal: 
        	            	expected: 201
        	            	actual  : 501
        	Test:       	TestStartProfiling
failed to write error response: applicationZip producer has not yet been implemented
    profiling_test.go:97: 
        	Error Trace:	profiling_test.go:97
        	Error:      	Not equal: 
        	            	expected: 200
        	            	actual  : 501
        	Test:       	TestStartProfiling
    profiling_test.go:103: 
        	Error Trace:	profiling_test.go:103
        	Error:      	Not equal: 
        	            	expected: "application/zip"
        	            	actual  : "text/plain; charset=utf-8"
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-application/zip
        	            	+text/plain; charset=utf-8
        	Test:       	TestStartProfiling
    profiling_test.go:107: 
        	Error Trace:	profiling_test.go:107
        	Error:      	Expected nil, but got: &errors.errorString{s:"zip: not a valid zip file"}
        	Test:       	TestStartProfiling
        	Messages:   	start/stop profiling returned an error: zip: not a valid zip file
--- FAIL: TestStartProfiling (5.01s)
```

**Comment by kaankabalak** [comment]
> Seems like the integration tests are failing, will take a look at it. Marking the PR as WIP for now.

Fixed the integration test failures, reopening the PR for reviews

File: .github/workflows/jobs.yaml [hunk 1]
```diff
@@ -1706,7 +1706,7 @@ jobs:
           go tool cover -func=all.out | grep total > tmp2
           result=`cat tmp2 | awk 'END {print $3}'`
           result=${result%\%}
-          threshold=53.40
+          threshold=53.30
```
Comment (dvaldivia) [inline .github/workflows/jobs.yaml:1709]: We shouldn't decrease the coverage threshold, if enough code was introduced to cause this to fail, additional tests need to be written to cover the new code.

In this case`ws_handle.go` code added has no test for it it seems
Screen Shot 2022-07-22 at 11 08 02 AM
Screen Shot 2022-07-22 at 11 07 58 AM

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: .github/workflows/jobs.yaml (hunk 1 above)
Comment (kaankabalak) [reply .github/workflows/jobs.yaml:1709]: Thanks for letting me know @dvaldivia, will add tests for this part of the code 👍

Comment (kaankabalak) [reply .github/workflows/jobs.yaml]: Fixed

**Comment by kaankabalak** [comment]
@dvaldivia, I have addressed your feedback, would appreciate if you could take another look. Thanks in advance 👍

**Review by dvaldivia — APPROVED** [review]
Tested. LGTM.

**Review by bexsoft — APPROVED** [review]
LGTM, Do we still need the Stop Profiling button?

**Comment by kaankabalak** [comment]
@bexsoft I thought we might want to keep it just in case the profiling takes too long. If required, I can send in another PR to make this change 👍

**Pull request merged**